src/ducks/resource.js

```diff
@@ -86,43 +56,31 @@ function resetTranslations() {
   };
 }
 
-function fetchTranslations(id, type, project, language) {
-  type = type || 'projects';
+function fetchTranslations(translated_id, type, project, language) {
+  const translated_type = type || 'project';
   return (dispatch) => {
     dispatch({
       type: FETCH_TRANSLATIONS,
       resource_type: type
     });
-    awaitTranslations(id, type, project)
+    apiClient.type('translations').get({ translated_type, translated_id })
     .then((resources) => {
       const { primary_language } = project;
       const { original, translations } = filterResources(resources, primary_language);
-      if (language && language.value !== primary_language) {
-        if (translations.length) {
-          dispatch(selectTranslation(original, translations, type, language));
-          dispatch({
-            type: FETCH_TRANSLATIONS_SUCCESS,
-            payload: { original, translations, loading: false }
-          });
-        } else {
-          awaitTranslations(id, type, project, language)
-          .then(([translation]) => {
-            if (translation) {
-              translations.push(translation);
-            }
-            dispatch(selectTranslation(original, translations, type, language));
-            dispatch({
-              type: FETCH_TRANSLATIONS_SUCCESS,
-              payload: { original, translations, loading: false }
-            });
-          });
-        }
-      } else {
-        dispatch({
-          type: FETCH_TRANSLATIONS_SUCCESS,
-          payload: { original, translations, loading: false }
-        });
+      if (translations.length && language) {
+        dispatch(selectTranslation(original, translations, type, language));
       }
+      dispatch({
+        type: FETCH_TRANSLATIONS_SUCCESS,
+        payload: { original, translations, loading: false }
+      });
+    })
+    .catch((error) => {
+      console.warn('Unable to fetch translations', error);
+      dispatch({
+        type: API_ERROR,
```

Comment:
Not a blocker, but this alters the idea of action type constants. If you want to have a generalised way of handling API calls, I thought [something like this](http://redux.js.org/docs/recipes/ReducingBoilerplate.html#async-action-creators)(Paragraph starting with "Finally, you can write your own middleware") would be better (although more code in the first place)

Reply:
I think I'm going to use a helper function, like the error handler in zooniverse/edu-api-frontend

Reply:
Updated in c4e1d87 to pass errors to a handler function.